app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CoursesController < ApplicationController
+  def index
```

Comment:
please use before action for finding the course for edit, update, destroy, and show.